src/org/intellij/erlang/completion/QuoteInsertHandler.java

```diff
@@ -65,13 +71,24 @@ public static void process(@NotNull Project project, @NotNull String name, @Null
       context.commitDocument();
       int tailOffset = context.getTailOffset();
       int startOffset = moduleName == null ? context.getStartOffset() : tailOffset - name.length();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(tailOffset + 1, QUOTA);
+
+      if (!alreadyHasQuote(context, startOffset - 1)) {
+        document.insertString(startOffset, QUOTA);
+      }
+      if (!alreadyHasQuote(context, tailOffset)) {
+        document.insertString(tailOffset + 1, QUOTA);
+      }
+
       editor.getCaretModel().moveToOffset(tailOffset + 2);
       context.setTailOffset(tailOffset + 2);
     }
   }
 
+  private static boolean alreadyHasQuote(@NotNull InsertionContext context, int position) {
+    PsiElement mayBeQuote = context.getFile().findElementAt(position);
+    return mayBeQuote != null && QUOTA.equals(mayBeQuote.getText());
```

Comment:
Do not use getText on PSI elements unless you really need it. In this case you can compare element type instead.